### DISCUSSION TITLE
Open Suse Tumbleweed Repository

### DISCUSSION THREAD

**Discussion by DanPartelly** [ipspace/netlab#1316] [General]
Since I worked a lot with networklab in the last weeks, I added a Tumbleweed repository here for networklab itself, and other support programs, like an up to date containerlab, ansible support for libssh, and the cpu-checker. 

https://build.opensuse.org/project/show/home:danp

Networklab wise, for now it offers **networklab-latest** RPM package which is built on demand from latest dev branch commit available at the time of triggering the build. the versioning of this package is simple:  date-short commit hash-revision. Ex: 20240913.77aefb8-5.1.  So you can always easily tell how out of date the package is. 

 I will add a **networklab** package which will follow the stable release_X.Y.Z.

Package maturity is low. for now, i completely skip any testing of networklab during build. Examples are not copied into /usr/share hierarchy. And so on and so forth.  Things will improve in time. 

As a policy, **networklab-latest** will have triggered rebuild when I need a bleeding edge feature or I need to test code. The future **networklab**  package will be update on release_X.Y.Z (but not on  -postX and -devX).

If the repository structure and URL will change, Ill make a follow up post here. It is probable that it will do at some point.

so, if you use Suse family, you might find this useful. Or not. But it's there.

**Comment by ipspace** [comment]
This is great. Thanks a million! Could you add a short description in the `docs/install` directory (for example, `docs/install/suse.md`) and submit it as a PR, then I'll link to it from the main installation document.

Also, -postX releases are important. They contain bug fixes that I don't want to delay until the next full release.

**Reply by DanPartelly** [reply]
Sure, Ill make the .md as soon as the package for stable networklab is ready and published. Ill take care of -post then, too. 
> 
> Also, -postX releases are important. They contain bug fixes that I don't want to delay until the next full release.
